CodingChallenges/CC_90_dithering_p5.js/sketch.js

```diff
@@ -0,0 +1,93 @@
+let kitten;
+
+function preload() {
+  kitten = loadImage("data/kitten.jpg");
+}
+
+function setup() {
+  createCanvas(1024, 512);
+
+  image(kitten, 0, 0);
+  makeDithered(kitten, 1);
+  image(kitten, 512, 0);
+  // Apply gray filter to the whole canvas
+  filter(GRAY);
+}
+
+function imageIndex(img, x, y) {
+  return 4 * (x + y * img.width);
+}
+
+function getColorAtindex(img, x, y) {
+  const idx = imageIndex(img, x, y);
+  const pix = img.pixels;
+  const red = pix[idx];
+  const green = pix[idx + 1];
+  const blue = pix[idx + 2];
+  const alpha = pix[idx + 3];
+  return color(red, green, blue, alpha);
```

Comment:
Ah, maybe it's good to use the color objects since you are returning something here. . . . this is a good way to organize the code for sure, but would it be simpler for a user to understand if this code were inside the for loop itself. I think my comments should actually be ignored and we should merge this thank you so so much for the help!